server/src/main/java/org/opensearch/action/admin/cluster/stats/ClusterStatsIndices.java

```diff
@@ -78,26 +81,44 @@ public ClusterStatsIndices(List<ClusterStatsNodeResponse> nodeResponses, Mapping
         this.segments = new SegmentsStats();
 
         for (ClusterStatsNodeResponse r : nodeResponses) {
-            for (org.opensearch.action.admin.indices.stats.ShardStats shardStats : r.shardsStats()) {
-                ShardStats indexShardStats = countsPerIndex.get(shardStats.getShardRouting().getIndexName());
-                if (indexShardStats == null) {
-                    indexShardStats = new ShardStats();
-                    countsPerIndex.put(shardStats.getShardRouting().getIndexName(), indexShardStats);
-                }
-
-                indexShardStats.total++;
-
-                CommonStats shardCommonStats = shardStats.getStats();
-
-                if (shardStats.getShardRouting().primary()) {
-                    indexShardStats.primaries++;
-                    docs.add(shardCommonStats.docs);
+            // Aggregated response from the node
+            if (r.getAggregatedNodeLevelStats() != null) {
+                r.getAggregatedNodeLevelStats().indexStatsMap.forEach(
```

Comment:
In ClusterStatsIndices, we create a `countsPerIndex` which has a cluster level accumulated information on each index and its ShardStats (accumulated over each Shard from each node belonging to that index)

To handle multi-version scenarios where some nodes in older versions may not have the `AggregatedNodeLevelStats` in their response we do the following:

1. If we do have `AggregatedNodeLevelStats`, we merge the results into the overall `countsPerIndex` map. This is done so that stats of the same index from different nodes are added up.
2. If we do not have `AggregatedNodeLevelStats`, we use the default method of adding the shardStats by iterating over each `indices.stats.ShardStats` returned from each node.